main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeArrivalAndDepartureTimeValidatorTest.java

```diff
@@ -0,0 +1,353 @@
+import org.mobilitydata.gtfsvalidator.notice.StopTimeWithArrivalBeforePreviousDepartureTimeNotice;
+import org.mobilitydata.gtfsvalidator.notice.StopTimeWithDepartureBeforeArrivalTimeNotice;
+import org.mobilitydata.gtfsvalidator.notice.StopTimeWithOnlyArrivalOrDepartureTimeNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableLoader;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class StopTimeArrivalAndDepartureTimeValidatorTest {
+    @Mock
+    final GtfsStopTimeTableContainer mockStopTimeTable = mock(GtfsStopTimeTableContainer.class);
+    @InjectMocks
+    final StopTimeArrivalAndDepartureTimeValidator underTest = new StopTimeArrivalAndDepartureTimeValidator();
+
+    @Before
+    public void openMocks() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void departureTimeAndArrivalTimeNotProvidedShouldNotGenerateNotice() {
+        NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
+        ListMultimap<String, GtfsStopTime> stopTimeCollection = ArrayListMultimap.create();
+        GtfsStopTime mockStopTime = mock(GtfsStopTime.class);
+        when(mockStopTime.hasArrivalTime()).thenReturn(false);
+        when(mockStopTime.hasDepartureTime()).thenReturn(false);
+        stopTimeCollection.put("trip id value", mockStopTime);
+        when(mockStopTimeTable.byTripIdMap()).thenReturn(stopTimeCollection);
+
+        underTest.validate(mockNoticeContainer);
+
+        verifyNoInteractions(mockNoticeContainer);
+        verify(mockStopTimeTable, times(1)).byTripIdMap();
+    }
+
+    @Test
+    public void departureTimeAfterArrivalTimeShouldNotGenerateNotice() {
+        NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
+        ListMultimap<String, GtfsStopTime> stopTimeCollection = ArrayListMultimap.create();
+        GtfsStopTime mockStopTime = mock(GtfsStopTime.class);
+        when(mockStopTime.hasArrivalTime()).thenReturn(true);
+        GtfsTime mockArrivalTime = mock(GtfsTime.class);
+        when(mockStopTime.arrivalTime()).thenReturn(mockArrivalTime);
+        when(mockStopTime.hasDepartureTime()).thenReturn(true);
+        GtfsTime mockDepartureTime = mock(GtfsTime.class);
+        when(mockStopTime.departureTime()).thenReturn(mockDepartureTime);
+        when(mockDepartureTime.isBefore(mockArrivalTime)).thenReturn(false);
```

Comment:
Is the behavior of `isBefore()` tested anywhere? If not we should have separate tests for this, or instead of mocking `GtfsTime` we can instantiate a real object with real values. Same for `isAfter()`.

Reply:
`isBefore()` is not tested yet but will be.

Reply:
These test will be included in https://github.com/MobilityData/gtfs-validator/pull/568.